cmd/contour/serve.go

```diff
@@ -375,36 +386,64 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 
 	// Wrap eventHandler in a converter for objects from the dynamic client.
 	// and an EventRecorder which tracks API server events.
-	dynamicHandler := k8s.DynamicClientHandler{
+	dynamicHandler := &k8s.DynamicClientHandler{
 		Next: &contour.EventRecorder{
 			Next:    eventHandler,
 			Counter: contourMetrics.EventHandlerOperations,
 		},
-		Converter: converter,
-		Logger:    log.WithField("context", "dynamicHandler"),
+		Logger: log.WithField("context", "dynamicHandler"),
 	}
 
-	// Inform on DefaultResources.
-	for _, r := range k8s.DefaultResources() {
-		inf, err := clients.InformerForResource(r)
-		if err != nil {
-			log.WithError(err).WithField("resource", r).Fatal("failed to create informer")
-		}
+	err = contour_api_v1.AddToScheme(mgr.GetScheme())
```

Comment:
It's a bit disappointing to end up moving back to doing everything individually, rather than being able to use a loop, I'd really like us to be making `doServe` smaller.

Maybe this is a good candidate for pulling out into a helper function? Feels like we could eliminate a bunch of boilerplate here.

Reply:
There actually is already a function to do this, so I swapped in the scheme that comes from the controller-runtime and can remove this. =)